### PR TITLE
Pwa zkorum

### PR DESCRIPTION
**Pull request by Balghisa12**
This pull request introduces Workbox, a set of JavaScript libraries, to enhance our web application with Progressive Web App (PWA) features. A service worker has been integrated using Workbox, allowing us to implement advanced caching strategies and improve offline capabilities.

Changes Made:

    Workbox Integration: Added Workbox library to leverage its powerful tools for building PWAs.
    Service Worker Implementation: Introduced a service worker to the project using Workbox, enabling offline caching and background sync.
    Caching Strategies: Configured a default caching strategy (NetworkFirst) to improve the user experience by prioritizing network requests and falling back to cached content when offline.
    Precaching: Leveraged Workbox's precacheAndRoute to automatically precache important assets, ensuring a fast and reliable initial load.

Benefits:

    Offline Availability: Users can now access the application even when offline, improving user experience in low or no connectivity scenarios.
    Faster Load Times: The implemented caching strategies enhance the loading speed by fetching resources from the cache when available, reducing network dependency.
    Improved Reliability: With Workbox, the service worker provides a robust solution for handling various scenarios, such as network failures and intermittent connectivity.

Testing:

    Unit Tests: Ensured that the integration of Workbox and the service worker doesn't break existing functionality.
    Offline Testing: Confirmed that the application works seamlessly in offline mode, and cached content is served appropriately.

Next Steps:

    Explore additional Workbox features to optimize and customize the caching strategies based on specific application requirements.
    Consider implementing background sync for improved data synchronization when the user is back online.

Reviewers:
@nicobao 

Relates to https://github.com/zkorum/zkorum/issues/25

### CODE REVIEW

File: services/front/src/main.tsx [hunk 1]
```diff
@@ -17,6 +17,36 @@ import { MainLoading } from "./MainLoading";
 import { Buffer } from "buffer";
 globalThis.Buffer = Buffer;
 
+navigator.serviceWorker
+ .register(
+   process.env.NODE_ENV === 'production' ? '/sw.js' : '/dev-sw.js?dev-sw'
```
Comment (nicobao) [inline services/front/src/main.tsx:22]: where is dev-sw.js?

File: services/front/vite.config.ts [hunk 2]
```diff
@@ -6,11 +6,31 @@ import checker from "vite-plugin-checker";
...
 import { VitePWA } from "vite-plugin-pwa";
 import { NodeGlobalsPolyfillPlugin } from "@esbuild-plugins/node-globals-polyfill";
-
 export default defineConfig({
     plugins: [
         react(),
-        VitePWA({ registerType: "autoUpdate", selfDestroying: true }),
+        VitePWA({
+            strategies: 'injectManifest', injectManifest: {
+                maximumFileSizeToCacheInBytes: 10000000
```
Comment (nicobao) [inline services/front/vite.config.ts:14]: is that how you solved the max file chunk issue?

File: services/front/package.json [hunk 3]
```diff
@@ -5,18 +5,18 @@
...
         "format:write": "prettier . --write",
         "preview": "vite preview",
         "test": "NODE_OPTIONS=--experimental-vm-modules jest",
         "image:build": "docker build -f Dockerfile.prod -t zkorum-front .",
-        "image:build:staging1": "docker build -f Dockerfile.staging1 -t zkorum-front-staging1 .",
-        "image:tag": "docker tag zkorum-frontJustForExample:latest zkorum/frontJustForExample:0.1.0",
-        "image:push": "docker image push zkorum/frontJustForExample:0.1.0"
+        "image:build:staging1": "docker build --no-cache -f Dockerfile.staging1 -t zkorum-front:haz .",
+        "image:tag": "docker tag zkorum-front:haz balghisa/zkorum:haz",
+        "image:push": "docker image push balghisa/zkorum:haz"
```
Comment (nicobao) [inline services/front/package.json:19]: rollback changes

File: services/front/package.json [hunk 4]
```diff
@@ -5,18 +5,18 @@
     "description": "ZKorum frontend",
     "scripts": {
         "dev": "vite",
-        "dev:force": "vite --force",
-        "build": "tsc && vite build",
-        "build:staging1": "tsc && NODE_ENV=development vite build --mode staging1",
+        "build:staging1": "tsc && NODE_ENV=production vite build --mode staging1",
+        "dev:force": "NODE_ENV='development' && vite --force",
+        "build": "NODE_ENV='production' && tsc && vite build",
```
Comment (nicobao) [inline services/front/package.json:10]: rollback changes:
- NODE_ENV=development or production is set automatically when using `vite` or `vite build`

File: services/front/package.json [hunk 5]
```diff
@@ -5,18 +5,18 @@
     "description": "ZKorum frontend",
     "scripts": {
         "dev": "vite",
-        "dev:force": "vite --force",
-        "build": "tsc && vite build",
-        "build:staging1": "tsc && NODE_ENV=development vite build --mode staging1",
+        "build:staging1": "tsc && NODE_ENV=production vite build --mode staging1",
```
Comment (nicobao) [inline services/front/package.json:8]: you can keep this, NODE_ENV=production makes sense, but put this line after "build" command

**Review by nicobao — CHANGES_REQUESTED** [review]

File: services/front/src/main.tsx (hunk 1 above)
Comment (nicobao) [reply services/front/src/main.tsx:22]: why do we need a separate file for dev?

**Comment by nicobao** [comment]
Great contribution, thank you <3
After you fix the small changes requested in the review, can you sign the FLA and add your full name and email address in the AUTHORS file so I can merge?

File: services/front/src/main.tsx (hunk 1 above)
Comment (Balghisa12) [reply services/front/src/main.tsx:22]: https://vite-pwa-org.netlify.app/guide/development.html#injectmanifest-strategy

Comment (Balghisa12) [reply services/front/src/main.tsx:22]: Everything is explained in this link

Comment (nicobao) [reply services/front/src/main.tsx:22]: According to the documentation we must use:
```ts
{
      devOptions: {
        enabled: true,
        type: "module",
      }
}
```
as well

File: README.md [hunk 6]
```diff
@@ -6,7 +6,7 @@ For some background, check the [organization page](https://github.com/zkorum) or
 
 ### Prerequisites
 
-Install:
+Install:d
```
Comment (nicobao) [inline README.md:9]: rollback

File: services/front/src/main.tsx (hunk 1 above)
Comment (nicobao) [reply services/front/src/main.tsx:22]: have you tested in dev mode?

File: services/front/vite.config.ts (hunk 2 above)
Comment (nicobao) [reply services/front/vite.config.ts:14]: ?

Comment (Balghisa12) [reply services/front/vite.config.ts:14]: yes

File: services/front/src/main.tsx (hunk 1 above)
Comment (nicobao) [reply services/front/src/main.tsx:22]: OK I  merge we will see later.

**Comment by nicobao** [comment]
PR merged, thanks @Balghisa12 :+1:

**Pull request merged**